### PR TITLE
Setting Up Warning Emails 10mins before login time

### PR DESCRIPTION
**Pull request by hanshengzhao1993**

### CODE REVIEW

File: src/public/components/AttendanceTable.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import cron from 'node-cron';
+
+const AttendanceTable = () => {
+  return (
+    <div>
+      <h1> New Page</h1>
+      {
+        cron.schedule('* * * * * *', function(){
```
Comment (Jukejc) [inline src/public/components/AttendanceTable.js]: whats this file for

File: src/public/components/requests/students.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ const removeStudentData = async (studentData) => {
 
 const storeAttendanceRecord = async (classes, time) => {
   try {
+    console.log('STOREATTENDANCE RECORD TIME:   ', time);
```
Comment (Jukejc) [inline src/public/components/requests/students.js]: console.log

File: src/public/components/requests/classes.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
 import axios from 'axios';
 
+const getAttendanceRecordDate = async (queryData) => {
+  try {
+    const reponse = await axios.post('/getAttendanceRecordDate', queryData)
```
Comment (Jukejc) [inline src/public/components/requests/classes.js]: should be axios.get and using queries https://expressjs.com/en/api.html#req.query

File: src/server/db/ModelHelpers/attendanceHelpers.js [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
     const { classes, time } = req.body;
     await Attendance.storeRecords(classes, time);
+
+    let warningEmail = setInterval( ()=> {
+      let warningTime = moment(time).subtract(10, 'minute');
+      let currentTime = moment();
+      let currentTimeString = currentTime.format('h:mm');
+      let recordedTimeString = warningTime.format('h:mm');
+      if(currentTimeString === recordedTimeString) {
+        console.log('currenttime and recorded time are same');
```
Comment (Jukejc) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: console.log

File: src/server/db/ModelHelpers/attendanceHelpers.js [hunk 5]
```diff
@@ -1,11 +1,35 @@
...
+        const pendingStudents = Attendance.emailWarningStudents();
+        clearInterval(warningEmail);
+      }
+    }, 5000);
+
+    var absentInterval = setInterval( () => {
```
Comment (Jukejc) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: var

File: src/public/components/requests/students.js (hunk 2 above)
Comment (nguyenaiden) [inline src/public/components/requests/students.js]: https://cdn.meme.am/cache/instances/folder632/39942632.jpg

File: src/server/db/ModelHelpers/attendanceHelpers.js [hunk 6]
```diff
@@ -35,6 +59,15 @@ exports.emailLateStudents = async (req, res) => {
   }
 }
 
+exports.removeAttendanceRecordDate = async (req, res) => {
+  try {
+    await Attendance.deleteRecordDate(req.body);
+  } catch (err) {
+    console.log(err.message);
```
Comment (Jukejc) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: console.log

File: src/server/db/ModelHelpers/attendanceHelpers.js (hunk 4 above)
Comment (nguyenaiden) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: https://cdn.meme.am/cache/instances/folder632/39942632.jpg

File: src/public/components/requests/students.js [hunk 7]
```diff
@@ -2,7 +2,7 @@ import axios from 'axios';
 
 const storeStudentData = async (studentData) => {
   try {
-    const response = await axios.post('/studentUpload', studentData);
+    const response = await axios.post('/studentUpload', {date: studentData});
```
Comment (Jukejc) [inline src/public/components/requests/students.js]: would like to see how this works also

File: src/server/db/ModelHelpers/attendanceHelpers.js (hunk 5 above)
Comment (nguyenaiden) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: A wild var is spotted.

File: src/server/db/ModelHelpers/attendanceHelpers.js (hunk 6 above)
Comment (nguyenaiden) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: https://cdn.meme.am/cache/instances/folder632/39942632.jpg

**Comment by Jukejc** [comment]
Closing, complete duplicate of more recently submitted PR #81

**Pull request closed without merging**